plugins/woocommerce/src/Admin/Features/ProductBlockEditor/RedirectionController.php

```diff
@@ -70,18 +67,20 @@ protected function is_product_supported( $product_id ): bool {
 	 * feature `product_block_editor` is enabled.
 	 */
 	public function maybe_redirect_to_new_editor(): void {
-		if ( $this->is_legacy_add_new_screen() ) {
-			wp_safe_redirect( admin_url( 'admin.php?page=wc-admin&path=/add-product' ) );
-			exit();
-		}
+		if ( \Automattic\WooCommerce\Utilities\FeaturesUtil::feature_is_enabled( 'product_block_editor' ) ) {
```

Comment:
This can be reverted now and these conditions moved back to the constructor now that the redirect is in place, right?